### ISSUE TITLE
Train a Classifier

### ISSUE THREAD

**Issue by ricky5151** [davidsandberg/facenet#644]
May I know what is the difference between Train a classifier on LFW & Train a classifier on your own dataset?

**Comment by IcyFrequency** [comment]
It seemed like "Train a classifier on LFW" trains a model with neural nets, and "Train a classifier on your own dataset?" uses sci kit learn to train a classifier using SVC:
http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html

Where the SVC requires a few more facial portraits to achieve better classification characteristics; at least when I tried it.

However, I'm not certain and would also like a more thorough explanation.

Thanks

**Comment by ricky5151** [comment]
@IcyFrequency  Do you have any idea on how to obtain training curve (Accuracy against Step & VAL @ FAR against Step) when used either these 2 methods?

**Comment by MaartenBloemen** [comment]
@IcyFrequency Actually both "Train a classifier on LFW" and "Train a classifier on your own dataset" use Scikit SVC to train a classifier. They both train this classifier based on the embeddings provided by the facenet model.

SVC will often require a lot of data because SVC is a multiclass classification algorithm based on SVM and uses the "One-VS-One" approach. Think that for every class you have, there will be N(N-1)/2 hyperplanes. So when you have a lot of classes you will have a lot of hyperplanes which in turn means you will have a lot of support vectors. So depending on the number of classes you want to train on you will need to supply sufficient data if you want to use this approach for classification.

**Comment by IcyFrequency** [comment]
@MaartenBloemen Thanks for your response. I'm working on something that needs the classifier to only use one facial image per person. And was therefore looking at this:

https://github.com/ageitgey/face_recognition

@ricky5151 No, I haven't done any evaluation (which i should) yet. I'll post my results once I get time to measure performance.
